### DISCUSSION TITLE
Woolley SA-001_POW flash pins

### DISCUSSION THREAD

**Discussion by drgkt** [arendst/Tasmota#12469] [Q&A] [answered]
As you see pins are marked: ERX, ETX, 3V3, GND, IO2, DRX, DTX.
Do I use the ERX, ETX or the DRX, DTX pins for flashing?

![WoolleySa-001POW](https://user-images.githubusercontent.com/32744663/123239464-0e28fb00-d4e8-11eb-8b28-7abf224f54ad.jpg)

**Comment by barbudor** [comment, marked as answer]
What is this device ? Is that the same as https://templates.blakadder.com/SA-001.html ?
Can you see the ESP or ESP-module ?

ETX and ERX could mean EST-TX and ESP-TX but not knowing the device I have no idea what DRX and DTX could stand for
You'd better try to follow the track (with a continuity tester) to find out where both are going
You will also need to tie IO0 to GND, which may be possible by just pressing the button

**Comment by drgkt** [comment]
1.[This](https://ewelinkcommunity.net/device-lists/plugs/woolley-sa-001-eu-without-ops/) device has 2 chips on pictured board 1.ESP8266EX and 2.WINBOND 25Q32JVSIQ1747.
2.Saša Miličević who also pointed to https://templates.blakadder.com/SA-001.html does not know either.
3."...tie IO0 to GND" What does that mean? (newbie!)

**Reply by barbudor** [reply]
1. Photo does show anything.
2. I don't know Sasa neither
3. Read Tasmota doc on flashing

**Comment by drgkt** [comment]
1. More photos. The link was to show device.
![IMG_20210410_150243](https://user-images.githubusercontent.com/32744663/123252471-fefc7a00-d4f4-11eb-8a3c-d44710af0a5c.jpg)
![IMG_20210410_150138](https://user-images.githubusercontent.com/32744663/123252476-002da700-d4f5-11eb-8005-c5cc93edf513.jpg)

2. Sasa is the person that answers at blackadder.com
3. Doc link? Will tell me which pins to use? (E or D)?

**Reply by barbudor** [reply]
Doc link ?
Haven't you already read [Tasmota doc](https://tasmota.github.io/docs/) 5 times before asking questions ?

Flashing instructions : https://tasmota.github.io/docs/Getting-Started/
Pinout of ESP8266 : https://tasmota.github.io/docs/Pinouts/#esp8266-and-esp8285

GPIO0 (or IO0) is a ESP8266 pin that must be tied to GND when resetting the ESP (or powering it up) to enter the [bootloader/flasher mode](https://tasmota.github.io/docs/Getting-Started/#programming-mode). If the template is correct, GPIO0 seems to be connected to the device button. So you will have to press the button while powering the device through USB/Serial adapter.
You need the recommended [golden CH340](https://tasmota.github.io/docs/Getting-Started/#serial-to-usb-adapter) adapter which include a proper 3V3 regulator to power the device while flashing it. 

Up to you to use your eyes or your continuity tester to find out which of your edge connector pins (ERX, ETX, ...) are connected to U0RXD and U0TXD. Although guessing that ERX/ETX are the one is probably a very good bet.

**Comment by drgkt** [comment]
So the right pins are the ones that beep with chip legs 25 and 26. Correct?
I have done what you wrote prior to that, the edge pin IO2 confused me.
And I have this ready!
![CH340](https://user-images.githubusercontent.com/32744663/123257972-8220ce80-d4fb-11eb-9170-e1bc2e02174a.jpg)

**Reply by barbudor** [reply]
> So the right pins are the ones that beep with chip legs 25 and 26. Correct?

Yes

> I have done what you wrote prior to that, the edge pin IO2 confused me.

Ignore IO2. It's an available IO pin (but also could impact the boot mode if changed)

> And I have this ready!
> ![CH340](https://user-images.githubusercontent.com/32744663/123257972-8220ce80-d4fb-11eb-9170-e1bc2e02174a.jpg)

So GO ! 😂

**Reply by drgkt** [reply]
THANK YOUUUUUUUUU!

**Comment by drgkt** [comment]
How to mark this answered by barbudor?
